plugin.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Secret holds the parameters of the SecretProviderClass CRD. Links the GCP
+// secret resource name to a path in the filesystem.
+type Secret struct {
+	// ResourceName refers to a SecretVersion in the format
+	// projects/*/secrets/*/versions/*.
+	ResourceName string `json:"resourceName" yaml:"resourceName"`
+
+	// FileName is where the contents of the secret are to be written.
+	FileName string `json:"fileName" yaml:"fileName"`
+}
+
+// StringArray holds the 'secrets' key of the SecretProviderClass parameters.
+// This is an array of yaml strings. Each string then must be parsed as a
+// 'Secret' struct.
+type StringArray struct {
+	Array []string `json:"array" yaml:"array"`
```

Comment:
If you name this `array`, do you still need the tags?

Reply:
If it doesn't start with the capital letter, it can't be exported (it's private).